strawberry_django/fields/base.py

```diff
@@ -202,7 +207,10 @@ def resolve_type(
                 self.origin_django_type,
             )
             if is_optional(
-                model_field,
+                model_field.output_field  # type: ignore
+                if GeneratedField is not None
+                and isinstance(model_field, GeneratedField)
+                else model_field,
```

Comment:
nit: add parenthesis around those to avoid them formatting in the same indentation, which IMO makes it confusing to see that everything is basically a single argument

obs. might need to run ruff again as it might adjust this differently

```suggestion
                (
                    model_field.output_field  # type: ignore
                    if (
                        GeneratedField is not None
                        and isinstance(model_field, GeneratedField)
                    )
                    else model_field
                ),
```